interface/modules/zend_modules/module/CodeTypes/src/CodeTypes/Listener/CodeTypeEventsSubscriber.php

```diff
@@ -0,0 +1,207 @@
+use OpenEMR\Events\Codes\CodeTypeInstalledEvent;
+use OpenEMR\Events\Core\SQLUpgradeEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class CodeTypeEventsSubscriber implements EventSubscriberInterface
+{
+    private const SNOMED_ENCOUNTER_TYPE_MAPPINGS = [
+        'visit-after-hours' => '185463005',
+        'visit-after-hours-not-night' => '185464004',
+        'weekend-visit' => '185465003',
+        'office-visit' => '30346009',
+        'established-patient' => '3391000175108',
+        'new-patient' => '37894004',
+        'postoperative-follow-up' => '439740005'
+    ];
+
+    private const CODE_TYPE_SNOMED = "SNOMED";
+    private const CODE_TYPE_SNOMED_CT = "SNOMED-CT";
+    private const CODE_TYPE_SNOMED_PR = "SNOMED-PR";
+    private const CODE_TYPE_CPT4 = "CPT4";
+
+    private const CPT4_ENCOUNTER_TYPE_MAPPINGS = [
+        'new-patient-10' => 'New Patient (Brief)'
+        ,'new-patient-15-29' => 'New Patient (Limited)'
+        ,'new-patient-30-44' => 'Level 3, New Patient, Office Visit'
+        , 'new-patient-45-59' => 'Extended Physical Exam'
+        , 'new-patient-60-74' => 'New Exam (Comprehensive)'
+        , 'established-patient-10-19' => 'Established Patient (Limited)'
+        , 'established-patient-20-29' => 'Established Patient (Detailed)'
+        , 'established-patient-30-39' => 'Established Patient (Extended)'
+        , 'established-patient-40-54' => 'Established Patient (Comprehensive)'
+    ];
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            SQLUpgradeEvent::EVENT_UPGRADE_POST => 'onSqlUpgradeEvent'
+            ,CodeTypeInstalledEvent::EVENT_INSTALLED_POST => 'onCodeTypeInstalledEvent'
+        ];
+    }
+
+    public function onSqlUpgradeEvent(SQLUpgradeEvent $event)
+    {
+        // grab our currently installed code types, check for SNOMED-CT AND CPT4 and then update accordingly.
+        // SELECT * FROM code_types WHERE ct_active=1 ORDER BY ct_seq, ct_key
+        $activatedCodeTypes = QueryUtils::fetchRecords("SELECT ct_key FROM code_types WHERE ct_active=1 ORDER BY ct_seq, ct_key");
+        if (empty($activatedCodeTypes)) {
+            return;
+        }
+        // we want to push out to the system that we are making changes...
+        $logger = function ($message) use ($event) {
+            // make sure we escape this here.
+            $event->getSqlUpgradeService()->flush_echo(text($message) . "<br />");
+        };
+
+        foreach ($activatedCodeTypes as $record) {
+            if ($record['ct_key'] == self::CODE_TYPE_CPT4) {
+                if ($this->shouldUpdateCPT4Mappings()) {
+                    $logger("Updating " . $record['ct_key'] . " Mappings");
+                    $this->updateCPT4Mappings($logger);
+                } else {
+                    $logger("Skipping Section #updateCPT4Mappings");
+                }
+            } else if ($this->isSnomedCodeType($record['ct_key'])) {
```

Comment:
I know @sjpadgett loves `elseif` instead :), it's just a minor phpcs warning

really cool solution to this code set dilemma, great job!

Reply:
for php should be elseif. for JS will be else if. Both are acceptable unless colon shortcutting 
still safety says elseif...